dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_CONVOLUTION_CONVOLUTION_GPU_H_
+#define DALI_KERNELS_IMGPROC_CONVOLUTION_CONVOLUTION_GPU_H_
+
+#include "dali/core/boundary.h"
+#include "dali/core/convert.h"
+#include "dali/core/format.h"
+#include "dali/core/tensor_view.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/cutlass/device/gemm.h"
+#include "dali/kernels/kernel.h"
+#include "dali/pipeline/util/operator_impl_utils.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * @brief Apply convolution with 1-channel `window` in specified axis.
+ *
+ * Operation is done by using GEMM and generating the matrix from convolution window on the fly
+ */
+template <typename Out, typename In, typename W, int ndim, int axis, bool has_channels = true>
+struct ConvolutionGpu {
+  KernelRequirements Setup(KernelContext& ctx, const TensorListShape<ndim>& in_shape,
+                           const TensorListShape<1>& window_size) {
+    KernelRequirements req;
+    ScratchpadEstimator se;
+    DALI_ENFORCE(
+        in_shape.size() == window_size.size(),
+        make_string("Provided input shape and window sizes do not mach in number of samples: ",
+                    in_shape.size(), " vs ", window_size.size(), "."));
+    int num_samples = in_shape.size();
+    for (int i = 0; i < num_samples; i++) {
+      int num_channels = has_channels ? in_shape[i][ndim - 1] : 1;
+      DALI_ENFORCE(window_size[i][0] % 2 == 1,
+                   make_string("Kernel window should have odd length, got: ", window_size,
```

Comment:
That's an artificial constraint...?

Reply:
It's assumed to be symmetric and centered over output, the even one doesn't have this property.
If needed it can be either handled by 0-padding (for the most lazy solution) or extended, but I would rather put the priority in adding GaussianBlur GPU operator first.

Reply:
At the very least change the API to generic and scream that it's not implemented if the window doesn't conform. And yes, print a BIG FAT TODO to the output if this unsupported code path gets executed.

Reply:
Adjusted the error massage to say it's not yet supported rather than not supported.